custom/enikshay/data_store.py

```diff
@@ -32,10 +32,11 @@ def latest_adherence_date(self, episode_id):
         if result:
             # the result is sorted on 'adherence_date'
             latest_date = result[0].get('adherence_date')
-            if not latest_date or not parse_date(latest_date):
+            parsed_date = parse(latest_date).date()
```

Comment:
ok - depending on the source data it's usually still good to do validation first. for example, if the value given is a time string, it will just fill in the current date and not raise an error (`parse('02:00')`).

also if parse() raises an exception if a bad format is given will that cause an issue here?